sn_transfers/src/wallet/hot_wallet.rs

```
@@ -591,15 +595,31 @@ impl HotWallet {
     }
 
     /// Loads a serialized wallet from a path.
+    // TODO: what's the behaviour here if path has stored key and we pass one in?
     fn load_from_path_and_key(wallet_dir: &Path, main_key: Option<MainSecretKey>) -> Result<Self> {
-        let key = match get_main_key(wallet_dir)? {
-            Some(key) => key,
-            None => {
-                let key = main_key.unwrap_or(MainSecretKey::random());
-                store_new_keypair(wallet_dir, &key)?;
-                warn!("No main key found when loading wallet from path, generating a new one with pubkey: {:?}", key.main_pubkey());
+        let key = match get_main_key_from_disk(wallet_dir) {
+            Ok(key) => {
+                if let Some(passed_key) = main_key {
+                    if key.secret_key() != passed_key.secret_key() {
+                        warn!("main_key passed to load_from_path_and_key, but a key was found in the wallet dir. Using the one found in the wallet dir.");
```

sounds bit risky.
shall we raise an error and terminate the work instead ?
if user do forgot main_key, he can still continue by passing down a None ?

---

I was debating this, I think it's a follow up. Changing the internal load fn to not do this and error is like 80% of this PR, so not super keen to start in on more in this :D